examples/write_and_read_streams.cpp

```diff
@@ -0,0 +1,289 @@
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <sparrow_ipc/deserialize.hpp>
+#include <sparrow_ipc/memory_output_stream.hpp>
+#include <sparrow_ipc/serializer.hpp>
+
+#include <sparrow/record_batch.hpp>
+
+namespace sp = sparrow;
+
+// Random number generator
+std::random_device rd;
+std::mt19937 gen(rd());
+
+/**
+ * Helper function to create a record batch with the same schema but random values
+ * All batches have: int32 column, float column, bool column, and string column
+ */
+sp::record_batch create_random_record_batch(size_t num_rows)
```

Comment:
Maybe move the helper functions to another file to improve readability?

Reply:
I think moving them into a "detail" or "util" namespace should be enough.

Reply:
Fixed